### PR TITLE
MOTECH-2776: error message handle and parse to json

### PR DESCRIPTION
**Pull request by palbecki**
There was empty error message if you add a configuration to CommCare with an incorrect password. Now the returned http response is valid json.

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by matkwiatkowski** [comment]
test this please

File: commcare/src/main/java/org/motechproject/commcare/web/CommcareController.java [hunk 1]
```diff
@@ -56,7 +58,12 @@ public String handleMiscException(Exception e) {
 
     private String handleException(Exception e) {
         logger.error(e.getMessage(), e);
-        return e.getMessage();
+
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty("message", e.getMessage());
+
+        Gson gson = new Gson();
+        return gson.toJson(jsonObject);
```
Comment (matkwiatkowski) [inline commcare/src/main/java/org/motechproject/commcare/web/CommcareController.java]: Now when we change handleException method in commcare controller we should also change getting message value from response in javascript handlers as you did below for other ones. Please find all handlers that can be affected by your changes and fix it

**Comment by matkwiatkowski** [comment]
test this please

**Comment by matkwiatkowski** [comment]
@kmadej looks good to me

File: commcare/src/main/java/org/motechproject/commcare/web/CommcareController.java (hunk 1 above)
Comment (kmadej) [inline commcare/src/main/java/org/motechproject/commcare/web/CommcareController.java]: We should make change only in frontend, this type of handling exception in backend is bad

**Comment by palbecki** [comment]
I used MotechMessage from commons.api.json. Error handlers in other modules use custom error class, but Exceptions in Commcare have only message.

**Comment by kmadej** [comment]
Now it looks better, however please use handleException method as previous. In the future if we want to change sth, it will be easier to change it in one place

**Comment by jredlarski** [comment]
test this please

**Comment by kmadej** [comment]
test this please

**Comment by kmadej** [comment]
test this please

**Comment by kmadej** [comment]
test this please

**Pull request merged**